presto-hive/src/main/java/com/facebook/presto/hive/HivePartitionManager.java

```diff
@@ -225,7 +229,25 @@ private Table getTable(SemiTransactionalHiveMetastore metastore, SchemaTableName
                     filter.add(HivePartitionKey.HIVE_DEFAULT_DYNAMIC_PARTITION);
                 }
                 else if (value instanceof Slice) {
-                    filter.add(((Slice) value).toStringUtf8());
+                    Type type = domain.getType();
+                    Slice slice = (Slice) value;
+                    if (type instanceof CharType) {
+                        int length = ((CharType) type).getLength();
+                        StringBuilder builder = new StringBuilder(length);
+                        String sliceString = slice.toStringUtf8();
+                        builder.append(sliceString);
+                        for (int i = sliceString.length(); i < length; i++) {
+                            builder.append(' ');
+                        }
+
+                        filter.add(builder.toString());
+                    }
+                    else if (type instanceof DecimalType) {
+                        filter.add(new SqlDecimal(unscaledDecimalToBigInteger(slice), ((DecimalType) type).getPrecision(), ((DecimalType) type).getScale()).toString());
```

Comment:
I'm not sure. I was hesitant because it doesn't seem parallel to what happens with ShortDecimal below (which is just value.toString()), but I tried it out and it worked.  

Also, I think I'll change this to use `Decimals.toString(slice, scale)`.